File: SlidingWindow/3LongestSubstringWithoutDuplicates.cpp
#include<set>
#include<algorithm>
#include<string>
class Solution {
public:
    int lengthOfLongestSubstring(std::string s) {
        int l = 0 ;
        int res = 0;
        std::set<int> mySet;

        for(int r = 0 ; r < s.size(); r++)
        {
            while(mySet.find(s[r]) != mySet.end())
            {
                mySet.erase(s[l]);
                l++;
            }
            mySet.insert(s[r]);
            res = std::max(res, r - l + 1 );
        }
        return res;
    }
};

File: SlidingWindow/3LongestSubstringWithoutDuplicates.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l = 0
        res = 0
        mySet = set()

        for r in range(len(s)):
            while s[r] in mySet :
                mySet.remove(s[l])
                l += 1
            mySet.add(s[r])
            res = max(res, r - l + 1)
        return res